src/Illuminate/Support/ValidatedInput.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace Illuminate\Support;
+
+use ArrayIterator;
+use Illuminate\Contracts\Support\ValidatedData;
+use stdClass;
+
+class ValidatedInput implements ValidatedData
```

Comment:
Any specific reason this isn't an actual collection?

(The only difference is the `only` method working on nested data, which we could easily add to the collection's method)